File: src-angular/task-manager/src/app/domain/task-id.ts
/**
 * Strongly-typed identifier for Task entities.
 */
export class TaskId {
  private constructor(readonly value: string) {}

  static new(): TaskId {
    return new TaskId(crypto.randomUUID());
  }

  static from(value: string): TaskId {
    return new TaskId(value);
  }

  equals(other: TaskId): boolean {
    return this.value === other.value;
  }

  toString(): string {
    return this.value;
  }
}

File: src-kotlin/task-manager-domain/src/main/kotlin/com/example/taskmanager/domain/tasks/TaskId.kt
package com.example.taskmanager.domain.tasks

import java.util.UUID

/**
 * Strongly typed identifier for a [Task].
 */
@JvmInline
value class TaskId(val value: UUID) {
    override fun toString(): String = value.toString()

    companion object {
        fun new(): TaskId = TaskId(UUID.randomUUID())

        fun fromUuid(value: UUID): TaskId = TaskId(value)

        fun fromString(value: String): TaskId = TaskId(UUID.fromString(value))
    }
}
